src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -69,8 +70,10 @@ internal sealed class HttpConnectionPool : IDisposable
 
         // HTTP/1.1 connection pool
 
-        /// <summary>List of available HTTP/1.1 connections stored in the pool.</summary>
-        private readonly List<HttpConnection> _availableHttp11Connections = new List<HttpConnection>();
+        /// <summary>Stack of currently available HTTP/1.1 connections stored in the pool.</summary>
+        private readonly ConcurrentStack<HttpConnection> _http11Connections = new();
```

Comment:
As we talked about offline, there's an unfortunate (small) allocation as part of returning items to this. I'm curious whether the contention this PR is avoiding was because of a lock being used at all, or also the scope of the lock being taken. If this were replaced just with a `Stack<HttpConnection>` and a lock around just the Push/Pop, does that still make a material difference? I can see the answer being either yes or no, so I just want to make sure we confirmed.

Reply:
(with a few % error between runs)

| citrine-lin | RPS w/ 256 connections |
|-------------|------------------------|
| main        | 659                    |
| Monitor     | 676                    |
| SpinLock    | 740                    |
| Conc. Stack | 900                    |
| Conc. Queue | 893                    |
| main 8x32   | 891                    |

Where Monitor & SpinLock are using a regular `Stack`.